src/app/core/components/left-nav/left-nav.component.ts

```diff
@@ -75,8 +75,12 @@ export class LeftNavComponent implements OnInit, OnDestroy {
   }
 
   onSelectionChangedByIdx(e: { index: number }): void {
-    this.activeTabIndex = e.index;
-    this.dataSources[e.index].focus();
+    this.changeTab(e.index);
+  }
+
+  changeTab(index: number): void {
+    this.activeTabIndex = index;
+    this.dataSources[index].focus();
     this.switchThemeColor();
```

Comment:
`switchThemeColor` does not really switch theme color.
Put the `this.themeChange.emit(this.activeTabIndex === 2 ? 'dark' : null);` line directly here, it will be cleaner.

Reply:
Fixed